### ISSUE TITLE
Warn when ignoring the number of bytes written by `write()`

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#80997]
I tried the following code ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=b77b69a13a51ac4b787b9c25cc7c6207)):

```rust
fn main() {
    use std::io::Write;
    let mut v = vec![];
    v.write(b"abc").unwrap();
}
```

I expected to see a warning that I ignored the number of bytes written, because it could be less than the number I tried to write. Instead, no warning happened.

I think this is a limitation of `#[must_use]` currently - it can't be applied to nested types, only the top-level return type.

See also https://github.com/ziglang/zig/issues/7775.

**Comment by jonas-schievink** [comment]
https://rust-lang.github.io/rust-clippy/master/index.html#unused_io_amount
